src/withHook.tsx

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+type passedHook<T> = (props: React.Props<T>) => any;
```

Comment:
Because we're destructuring the return value and passing it as props this should probably restrict them to returning an object or nothing `type passedHook<T> = (props: React.Props<T>) => object | undefined;`